server-common/src/main/java/org/apache/kafka/server/common/DeleteRecordsPartitionStatus.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.common;
+
+
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.message.DeleteRecordsResponseData;
+import org.apache.kafka.common.protocol.Errors;
+import org.apache.kafka.server.metrics.KafkaMetricsGroup;
+import org.apache.kafka.server.purgatory.DelayedOperation;
+
+import com.yammer.metrics.core.Meter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+public class DeleteRecordsPartitionStatus {
+    private final long requiredOffset;
+    private final DeleteRecordsResponseData.DeleteRecordsPartitionResult responseStatus;
+    private volatile boolean acksPending;
+
+    public DeleteRecordsPartitionStatus(long requiredOffset,
+                                        DeleteRecordsResponseData.DeleteRecordsPartitionResult responseStatus) {
+        this.requiredOffset = requiredOffset;
+        this.responseStatus = responseStatus;
+        this.acksPending = false;
+    }
+
+    public boolean isAcksPending() {
+        return acksPending;
+    }
+
+    public void setAcksPending(boolean acksPending) {
+        this.acksPending = acksPending;
+    }
+
+
+    public DeleteRecordsResponseData.DeleteRecordsPartitionResult getResponseStatus() {
```

Comment:
It seems that the get prefix is rarely used in Kafka, should change to `responseStatus`